extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -212,6 +204,95 @@ export class InstrumentationService {
   }
 
   /**
+   * Creates listeners for visibility conditions or calls the callback if all
+   * the conditions are met.
+   * @param {!AnalyticsEventListenerDef} The callback to call when the event
+   *   occurs.
+   * @param {!JSONObject} config Configuration for instrumentation.
+   * @private
+   */
+  createVisibilityListener_(callback, config) {
+    if (!this.isVisibilitySpecValid_(config)) {
+      return;
+    }
+
+    // TODO(avimehta, #1297): Add all the listeners so that visibility
+    // conditions are monitored and callback is called when the conditions
+    // are met.
+    if (this.viewer_.isVisible()) {
```

Comment:
Thanks for pointing out. Since this PR is about visibilitySpec, I'll not touch the bug(unless you insist). I'll send out another PR with the fix for this bug.